api/v1beta1/tags.go

```diff
@@ -130,18 +130,6 @@ const (
 
 	// CommonRole describes the value for the common role.
 	CommonRole = "common"
-
-	// VMTagsLastAppliedAnnotation is the key for the machine object annotation
-	// which tracks the AdditionalTags in the Machine Provider Config.
-	// See https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
-	// for annotation formatting rules.
-	VMTagsLastAppliedAnnotation = "sigs.k8s.io/cluster-api-provider-azure-last-applied-tags-vm"
-
-	// RGTagsLastAppliedAnnotation is the key for the Azure Cluster object annotation
-	// which tracks the AdditionalTags for Resource Group which is part in the Azure Cluster.
-	// See https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
-	// for annotation formatting rules.
-	RGTagsLastAppliedAnnotation = "sigs.k8s.io/cluster-api-provider-azure-last-applied-tags-rg"
```

Comment:
Since this is currently part of the api/, it would technically be a breaking change to remove it. Although I don't know of anyone one using these consts as a library apart from our own usage, let's be cautious and avoid removing them. I would suggest adding the const as you did below and switching our own usage to the new const as you did, but leaving leaving the existing constants in v1alpha3, v1alpha4, and v1beta1 API folders, with a note on the v1beta1 API marking the constants as deprecated as such:

```
// DEPRECATED: use azure.RGTagsLastAppliedAnnotation instead. This constant will be removed in v1beta2.
RGTagsLastAppliedAnnotation = "sigs.k8s.io/cluster-api-provider-azure-last-applied-tags-rg"
```

And same thing for the `VMTagsLastAppliedAnnotation` const. No need to modify the v1alpha3 and v1alpha4 APIs since the current API version is v1beta1.

Reply:
> leaving the existing constants in v1alpha3, v1alpha4, and v1beta1 API folders,

Addressed via a43946f0c4bc47f63384312c813c1c1b5a5b6ae5, 811180051bb6711ecf792910f784838b955f555c, 12144ca26edb58b9c68e18d0795a4217314927c0

> a note on the v1beta1 API marking the constants as deprecated as such

Added via d3f351820344f2fff001ca93311c7e2a1ee4e4fc

Reply:
@CecileRobertMichon Does `Release Notes` needs to be updated too?

Reply:
Updated the Release notes.